RFM9x.device.lib.nut

```diff
@@ -0,0 +1,437 @@
+        local clear = cur & 0xf1; // exclude bits 3-1
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, clear | (CRTABLE[cr] << 1));
+    }
+    
+    // Enable RX payload CRC
+    function setRxPayloadCRC(state) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG2);
+        _writeReg(RFM9X_REG_MODEM_CONFIG2, (current & 0xfb) | (state ? 0x04 : 0x00));
+    }
+    
+    // Sets the bandwidth of the radio
+    function setBandwidth(bw) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, BWTABLE[bw] << 4 | (current & 0x0f));
+    }
+    
+    // Sets the mode of the radio. Options are constants: SLEEP, STANDBY, FSTX,
+    // TX, FSRX, RXCONTINUOUS, RXSINGLE, CAD
+    function setMode(mode) {
+        local current = _readReg(RFM9X_REG_OP_MODE);
+        _writeReg(RFM9X_REG_OP_MODE, mode | (current & 0xf8));
+    }
+        
+    // Clear all interrupt flags
+    function clearInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS, 0xff);
+    }
+    
+    // Mask all interrupts. Calling this method will prevent the interrupt
+    // pin from being asserted
+    function maskAllInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS_MASK, 0xff);
+    }
+    
+    // Enable a specific interrupt. Avaliable interrupts are in
+    // RFM9X_FLAGS
+    function enableInterrupt(bitnumber) {
+        local currentInterrupts = _readReg(RFM9X_REG_IRQ_FLAGS);
+        // Mask the interrupt by writing high, therefore enable by writing low
+        local newInterrupts = currentInterrupts & (~ (1 << bitnumber));
+        _writeReg(RFM9X_REG_IRQ_FLAGS_MASK, newInterrupts);
+    }
+
+    // Sets the max payload size of the FIFO buffer
+    function setMaxPayload(pl) {
+        pl = pl & 0xff;
+        _writeReg(RFM9X_REG_MAX_PAYLOAD_LENGTH, pl);
+    }
+    
+    // Sets the spreading factor of the radio
+    function setSpreadingFactor(sf) {
+        _writeReg(RFM9X_REG_MODEM_CONFIG2, (sf & 0x0f) << 4);
+    }
+    
+    // Sets the transmission frequency of the radio
+    function setFrequency(freq) {
+        setMode(RFM9X_SLEEP);
+        local reg_freq = (freq/RFM9X_FREQ_STEP).tointeger();
+        local freq_blob = blob(4); // big-endian blob
+        freq_blob[0] = RFM9X_REG_FREQUENCY_HIGH | RFM9X_WRITE_MASK;
+        freq_blob[1] = (reg_freq >> 16) & 0xFF;
+        freq_blob[2] = (reg_freq >> 8) & 0xFF;
+        freq_blob[3] = (reg_freq) & 0xFF;
+        
+        _csLow();
+        _spiModule.writeread(freq_blob);
+        _csHigh();
+    }
+
+    // Sets the receive handler. The receive handler should take two 
+    // parameters: an error message and the received data. The error message
+    // will either be "error" or "valid". If the message is "error", the
+    // received data parameter will be passed as null. 
+    function setReceiveHandler(handler) {
+        _receiveHandler = handler;
+    }
+
+    // Returns whether the last send operation is ongoing
+    function isSending() {
```

Comment:
Naming mismatch with documentation - docs say isDoneSending() - which would be the opposite boolean return value, so please check and document this and update all examples that use this.